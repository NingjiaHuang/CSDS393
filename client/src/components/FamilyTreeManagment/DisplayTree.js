import React, {Fragment, useState, useEffect} from 'react'
import mock_data from "./MOCK_DATA_TREE.json"

const DisplayTree = (cat) =>{
  const[searchTerm, setSearchTerm] = useState("")
  const[tree, setTree] = useState([]);
  const getTree = async() => {
      try{
          const response = await fetch(`http://localhost:4020/api/v1/gettree/${cat.certi_num}`) //should be get tree of a certain cat
          const jsonData = await response.json()
          setTree(jsonData);
      } catch (err){
          console.error(err.message);
      }
  }

    useEffect(() =>{
        getTree();
    }, [])

    return(
        <Fragment>
             {" "}
          <input type="text" className="form-control" 
          placeholder="search a family member here" 
          onChange={event => {setSearchTerm(event.target.value)}}
          />
          <table class="table mt-5 text-center">
              <thead>
                  <tr>
                      <th>Cat Name</th>
                      <th>Certificate number</th>
                      <th>Sire path</th>
                      <th>Dam path</th>
                  </tr>
              </thead>
              <tbody>
              {mock_data.filter((val) =>{  //if get all tree completed, change "mock_data" to tree
                  if (searchTerm == ""){
                    return val
                  } else if (val.cat_name.toLowerCase().includes(searchTerm.toLowerCase()) ){
                    return val
                  }
              }
              ).map((val, key) => (
                  <tr key={key}>
                      <td>{val.cat_id}</td>
                      <td>{val.cat_name}</td>
                      <td>{val.sire_path}</td>
                      <td>{val.dam_path}</td>
                      <td>
                      </td>
                  </tr>
              ))
              }
            </tbody>
        </table>
        </Fragment>
    )
}

export default DisplayTree;