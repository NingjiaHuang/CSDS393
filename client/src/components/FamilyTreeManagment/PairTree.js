import React, {useState} from 'react'
import { Button, Modal } from 'react-bootstrap';

const PairTree = () =>{
    const [info, setInfo] = useState({
        id:"",
        cat_name:"",
        sire_id:"",
        dam_id:""
    });

    const {id, cat_name, sire_id, dam_id} = info;
        
    const onChange = (e) => {
        setInfo({...info, [e.target.name]:e.target.value})
    }

    const [show, setShow] = useState(false);
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);

    const onSubmitForm = async(e) => {
        e.preventDefault();
        try{
            const body = {id, cat_name, sire_id, dam_id};
            console.log(body);
            const response = await fetch("http://localhost:4020/api/v1/cats/tree",{ // url needs update. need to add to breeding cat table
                method:"POST",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            });

            window.location="/family_tree";
        } catch(err){
            console.error(err.message);
        }
    }



    return(
        <div>
                <Button data-testid="pairBtn" variant="info" onClick={handleShow}>Pair</Button>
                <Modal show={show} onHide={handleClose}>
                    <Modal.Header closeButton>
                    <Modal.Title>Pair a kitten with parents</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        <form onSubmit={onSubmitForm}>
                            <label htmlFor='id'>Kitten certificate number: </label>
                            <input type="text" 
                            name="id"   
                            id='id' 
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Kitten name: </label>
                            <input 
                            type="text" 
                            name="cat_name"  
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Sire certificate number: </label>
                            <input 
                            type="text" 
                            name="sire_id"  
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Dam certificate number: </label>
                            <input 
                            type="text" 
                            name="dam_id"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>
                        </form>
                    </Modal.Body>
                    <Modal.Footer>
                    <Button variant="secondary" onClick={handleClose}>
                        Close
                    </Button>
                    <Button variant="warning" onClick={onSubmitForm}>
                        Pair
                    </Button>
                    </Modal.Footer>
                </Modal>
        </div>
    )
}


export default PairTree;