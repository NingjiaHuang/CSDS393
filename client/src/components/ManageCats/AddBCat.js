import React, {Fragment, useState} from 'react'
import { Button, Modal } from 'react-bootstrap';
import { MDBContainer } from 'mdbreact';

const AddBcat = ({setAuth}) =>{
    const [info, setInfo] = useState({
        cur_owner_cattery:"",
        cur_owner:"",
        certi_num:"",
        title:"",
        cat_reg_name:"",
        cat_name:"",
        breed:"",
        sex:"",
        birth_date:"",
        sire_name:"",
        dam_name:"",
        sale_status:"",
        retire_status:"",
    });

    const {cur_owner_cattery, cur_owner, certi_num, title, cat_reg_name, cat_name,
        breed, sex, birth_date, sire_name, dam_name, sale_status,retire_status} = info;
        
    const onChange = (e) => {
        setInfo({...info, [e.target.name]:e.target.value})
    }

    const [show, setShow] = useState(false);
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);

    const onSubmitForm = async(e) => {
        e.preventDefault();
        try{
            const body = {cur_owner_cattery, cur_owner, certi_num, title, cat_reg_name, cat_name,
                breed, sex, birth_date, sire_name, dam_name, sale_status, retire_status};

            const response = await fetch("http://localhost:4020/api/v1/cats/create_breed", { // url needs update. need to add to breeding cat table
                method: "POST",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            });

            window.location="/manage_cat_breeder";
        } catch(err){
            console.error(err.message);
        }
    }



    return(
        <div>
                <Button data-testid="showBtn" variant="info" onClick={handleShow} >Add a breeding cat</Button>
                <Modal show={show} onHide={handleClose}>
                    <Modal.Header closeButton>
                    <Modal.Title>Add a breeding cat</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        <form name="addcat" data-testid="addcat" onSubmit={onSubmitForm}>
                            <label htmlFor="cur_owner_cattery">Current owner cattery name: </label>
                            <input type="text" 
                            name="cur_owner_cattery"    
                            id="cur_owner_cattery"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="cur_owner">Current owner: </label>
                            <input 
                            type="text" 
                            name="cur_owner" 
                            id="cur_owner"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="certi_num">Certificate number: </label>
                            <input 
                            type="text" 
                            name="certi_num"  
                            id="certi_num"

                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="title">Title: </label>
                            <input 
                            type="text" 
                            name="title"  
                            id="title" 
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="cat_reg_name">Cat registered name: </label>
                            <input type="text" 
                            name="cat_reg_name"
                            id="cat_reg_name"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="cat_name">Cat preferred name: </label>
                            <input type="text" 
                            name="cat_name"   
                            id="cat_name"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="breed">Breed: </label>
                            <input type="text" 
                            name="breed"   
                            id="breed"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="sex">Sex: </label>
                            <input type="text" 
                            name="sex"   
                            id="sex"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="birth_date">Date of birth: </label>
                            <input type="text" 
                            name="birth_date"   
                            id="birth_date"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="sire_name">Sire: </label>
                            <input type="text" 
                            name="sire_name" 
                            id="sire_name"  
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="dam_name">Dam: </label>
                            <input type="text" 
                            name="dam_name" 
                            id="dam_name"  
                            className="form-control my-2"
                            onChange={e => onChange(e)}/> 

                            <label htmlFor="sale_status">Sale status: </label>
                            <input type="text" 
                            name="sale_status"  
                            id="sale_status" 
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="retire_status">Retire status: </label>
                            <input type="text" 
                            name="retire_status"
                            id="retire_status"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>
                        </form>
                    </Modal.Body>
                    <Modal.Footer>
                    <Button data-testid = "closebtn" variant="secondary" onClick={handleClose}>
                        Close
                    </Button>
                    <Button variant="warning" onClick={onSubmitForm}>
                        Add
                    </Button>
                    </Modal.Footer>
                </Modal>
        </div>
    )
}

export default AddBcat;