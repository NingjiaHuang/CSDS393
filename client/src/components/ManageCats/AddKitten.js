import React, { useState} from 'react'
import { Button, Modal } from 'react-bootstrap';

const AddKitten = ({setAuth}) =>{
    const [info, setInfo] = useState({
        cur_owner_cattery:"",
        cur_owner:"",
        certi_num:"",
        title:"",
        cat_reg_name:"",
        cat_name:"",
        breed:"",
        sex:"",
        birth_date:"",
        sire_name:"",
        dam_name:"",
        sale_status:"",
        weight:"",
        health_cond:"",
        vaccination_cond:""
    });

    const {cur_owner_cattery, cur_owner, certi_num, title, cat_reg_name, cat_name,
        breed, sex, birth_date, sire_name, dam_name, sale_status, weight, health_cond, vaccination_cond} = info;
    
    const onChange = (e) => {
        setInfo({...info, [e.target.name]:e.target.value})
    }
    
    const [show, setShow] = useState(false);
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
    const onSubmitForm = async(e) => {
        e.preventDefault();
        try{
            const body = {cur_owner_cattery, cur_owner, certi_num, title, cat_reg_name, cat_name,
                breed, sex, birth_date, sire_name, dam_name, sale_status, weight, health_cond, vaccination_cond};
            const response = await fetch("http://localhost:4020/api/v1/cats/create_kitten",{
                method:"Post",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            });
            window.location="/manage_cat_breeder"
        } catch(err){
            console.error(err.message);
        }
    }



    return(
        <div>
                <Button data-testid="showBtn" variant="info" onClick={handleShow}>Add a kitten</Button>
                <Modal show={show} onHide={handleClose}>
                    <Modal.Header closeButton>
                    <Modal.Title>Add a kitten</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        <form>
                        <label htmlFor="cur_owner_cattery">Current owner cattery name: </label>
                            <input type="text" 
                            name="cur_owner_cattery"    
                            id="cur_owner_cattery"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="cur_owner">Current owner: </label>
                            <input 
                            type="text" 
                            name="cur_owner" 
                            id="cur_owner"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="certi_num">Certificate number: </label>
                            <input 
                            type="text" 
                            name="certi_num"  
                            id="certi_num"

                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="title">Title: </label>
                            <input 
                            type="text" 
                            name="title"  
                            id="title" 
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="cat_reg_name">Cat registered name: </label>
                            <input type="text" 
                            name="cat_reg_name"
                            id="cat_reg_name"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="cat_name">Cat preferred name: </label>
                            <input type="text" 
                            name="cat_name"   
                            id="cat_name"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="breed">Breed: </label>
                            <input type="text" 
                            name="breed"   
                            id="breed"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="sex">Sex: </label>
                            <input type="text" 
                            name="sex"   
                            id="sex"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="birth_date">Date of birth: </label>
                            <input type="text" 
                            name="birth_date"   
                            id="birth_date"
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="sire_name">Sire: </label>
                            <input type="text" 
                            name="sire_name" 
                            id="sire_name"  
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="dam_name">Dam: </label>
                            <input type="text" 
                            name="dam_name" 
                            id="dam_name"  
                            className="form-control my-2"
                            onChange={e => onChange(e)}/> 

                            <label htmlFor="sale_status">Sale status: </label>
                            <input type="text" 
                            name="sale_status"  
                            id="sale_status" 
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="weight">Current weight: </label>
                            <input type="text" 
                            id="weight"
                            name="weight"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="health_cond">Health condition: </label>
                            <input type="text" 
                            name="health_cond"  
                            id="health_cond"    
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label htmlFor="vaccination_cond">Vaccination condition: </label>
                            <input type="text" 
                            name="vaccination_cond"  
                            id="vaccination_cond"    
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>
                        </form>
                    </Modal.Body>
                    <Modal.Footer>
                    <Button variant="secondary" data-testid="closebtn" onClick={handleClose}>
                        Close
                    </Button>
                    <Button variant="warning" onClick={onSubmitForm}>
                        Add
                    </Button>
                    </Modal.Footer>
                </Modal>
        </div>
    )
}

export default AddKitten;