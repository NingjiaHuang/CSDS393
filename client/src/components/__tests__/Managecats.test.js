import {render, screen, cleanup} from '@testing-library/react';
import DisplayTree from '../FamilyTreeManagment/DisplayTree';
import PairTree from '../FamilyTreeManagment/PairTree';
import AddBCat from '../ManageCats/AddBCat';
import AddPCat from '../ManageCats/AddPCat';
import AddKitten from '../ManageCats/AddKitten';

import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';
afterEach(() =>{
    cleanup();
})

test('matches snapshot of add breeding cat', () =>{
    const tree = renderer.create(<AddBCat/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('matches snapshot of add pregnant cat', () =>{
    const tree = renderer.create(<AddPCat/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('matches snapshot of add a kitten', () =>{
    const tree = renderer.create(<AddKitten/>).toJSON();
    expect(tree).toMatchSnapshot();
})