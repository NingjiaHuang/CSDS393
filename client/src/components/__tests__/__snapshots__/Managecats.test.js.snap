// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of add a kitten 1`] = `
<div>
  <button
    className="btn btn-info"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add a kitten
  </button>
</div>
`;

exports[`matches snapshot of add breeding cat 1`] = `
<div>
  <button
    className="btn btn-info"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add a bredding cat
  </button>
</div>
`;

exports[`matches snapshot of add pregnant cat 1`] = `
<div>
  <button
    className="btn btn-info"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add a pregnant cat
  </button>
</div>
`;
